prompt_toolkit/formatted_text/base.py

```diff
@@ -21,9 +22,11 @@
     Tuple[str, str], Tuple[str, str, Callable[[MouseEvent], None]]
 ]
 
-# List of (style, text) tuples.
+# List (covariant version of list) of (style, text) tuples
+# Not there is a subclass of this 
 StyleAndTextTuples = List[OneStyleAndTextTuple]
-
+# For typing stuff, we prefer an Iterable as it is more flexible
+StyleAndTextTuplesType = Iterable[OneStyleAndTextTuple]
```

Comment:
I'm not sure that ``Iterable`` is desirable, because it means it's not indexable and possibly, can only be consumed once.
Without reading again through all of the prompt_toolkit codebase, I think ``Sequence`` is at least what we need.